### ISSUE TITLE
Could not load System.Private.ServiceModel, System.Reflection.MetadataLoadContext

### ISSUE THREAD

**Issue by DevDrake** [dotnet/runtime#36695]
### Description

dotnet core project version 3.1.202 that loads .net framework 4.6.1 and netstandard 2.0 libraries and reading all classes, methods and its metadata.
The problem exists in release **compilation**, where in **debug** works just ok. 

Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Private.ServiceModel, Version=4.5.0.3, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.  

maybe related: https://github.com/dotnet/wcf/issues/2824

### Configuration

Windows, anycpu, release

**Comment by DevDrake** [comment]
Maybe related: https://github.com/dotnet/sdk/issues/10718

**Comment by steveharter** [comment]
@HongGit @mconnew are there versioning or packaging issues with `System.Private.ServiceModel`?

@DevDrake do you know what triggered the load of `System.Private.ServiceModel, Version=4.5.0.3`? Are there `System.ServiceModel.*` assembly or package references? (I don't expect `System.Private.ServiceModel` to be referenced directly, but instead by other `System.ServiceModel.*` packages).

The issue is not likely in MetadataLoadContext but instead either an incorrect assembly reference or a version mismatch. However, I'm not sure why the issue doesn't repro in release mode.

**Comment by mconnew** [comment]
No versioning or package issues. The problem is most like with your application. I suspect you are  basically doing something like this:  
Application depends on  
    LibraryProject  
    System.ServiceModel.Primitives 4.4.x (any version < 4.5.0.3)  

LibraryProject depends on  
    System.ServiceModel.Primitives 4.5.0.3  

What happens is the application dependency overrides any transitive dependencies and forces the version to 4.4.x. LibraryProject has been compiled against 4.5.0.3 and at runtime it doesn't have a version at least 4.5.0.3 so can't load WCF.

An alternative is there might be a problem with bin placing libraries in the output folder in your project system. WCF doesn't do anything different between release and debug so this issue will be outside of the WCF packages.

**Comment by DevDrake** [comment]
@steveharter You are right, the problem is not related to MetadataLoadContext as i previously suspected.

The problem was introduced by calling `atypefromMLC  == typeof(OperationContractAttribute)` which I had to change to string comparison anyway, as the Type i get from MetadataLoadContext was from different framework, so basically different type.

@mconnew, the project was converted form .net framework 4.6.1 to dotnet core 3.1 by hand, I found that the System.ServiceModel was referenced as `<Reference>` not `<PackageReference>` in the csproj so that may be the case. However VS InelliSense was happy all the time so I wasynt paying much attention on the csproj source.
I did not dig it further as I removed the need for ServiceModel completely on the project side before.
If I get some free time i will try to confirm this.
